exporter/collector/metrics_integration_test.go

```diff
@@ -96,56 +93,72 @@ func TestIntegrationMetrics(t *testing.T) {
-				testServer.Shutdown()
-				require.NoError(t, exporter.Shutdown(ctx))
-				require.NoError(t, testServerExporter.Shutdown(ctx))
-			}()
-
 			metrics := test.LoadOTLPMetricsInput(t, startTime, endTime)
-			expectedCreateMetricDescriptorReq := test.LoadCreateMetricDescriptorFixture(
-				t,
-				startTime,
-				endTime,
-			)
-			expectedCreateTimeSeriesReq := test.LoadCreateTimeSeriesFixture(t, startTime, endTime)
-
-			// First try exporting to local GCM test server and compare the requests
-			require.NoError(
-				t,
-				testServerExporter.ConsumeMetrics(ctx, metrics),
-				"Failed to export metrics to local test server",
-			)
-			actualCreateMetricDescriptorReq := <-testServer.CreateMetricDescriptorChan
-			diff := integrationtest.DiffProtos(
-				actualCreateMetricDescriptorReq,
-				expectedCreateMetricDescriptorReq,
-			)
-			assert.Emptyf(
-				t,
-				diff,
-				"Expected CreateMetricDescriptor request and actual GCM request differ:\n%v",
-				diff,
-			)
-			actualCreateTimeSeriesReq := <-testServer.CreateTimeSeriesChan
-			diff = integrationtest.DiffProtos(
-				actualCreateTimeSeriesReq,
-				expectedCreateTimeSeriesReq,
-			)
-			assert.Emptyf(
-				t,
-				diff,
-				"Expected CreateTimeSeries request and actual GCM request differ:\n%v",
-				diff,
-			)
-			// Try exporting with the real GCM exporter
-			require.NoError(t, exporter.ConsumeMetrics(ctx, metrics), "Failed to export metrics")
+
+			mockExportPassed := t.Run("Mock Export", func(t *testing.T) {
+				testServer, err := integrationtest.NewMetricTestServer()
+				require.NoError(t, err)
+				go testServer.Serve()
+				testServerExporter := createMetricsTestServerExporter(ctx, t, testServer)
+
+				defer func() {
+					testServer.Shutdown()
+					require.NoError(t, testServerExporter.Shutdown(ctx))
+				}()
+
+				// First try exporting to local GCM test server and compare the requests
+				require.NoError(
+					t,
+					testServerExporter.ConsumeMetrics(ctx, metrics),
+					"Failed to export metrics to local test server",
+				)
+				actualCreateMetricDescriptorReq := <-testServer.CreateMetricDescriptorChan
+				actualCreateTimeSeriesReq := <-testServer.CreateTimeSeriesChan
+
+				expectedCreateMetricDescriptorReq := test.LoadCreateMetricDescriptorFixture(
+					t,
+					startTime,
+					endTime,
+				)
+				diff := integrationtest.DiffProtos(
+					actualCreateMetricDescriptorReq,
+					expectedCreateMetricDescriptorReq,
+				)
+				require.Emptyf(
+					t,
+					diff,
+					"Expected CreateMetricDescriptor request and actual GCM request differ:\n%v",
+					diff,
+				)
+				expectedCreateTimeSeriesReq := test.LoadCreateTimeSeriesFixture(t, startTime, endTime)
+				diff = integrationtest.DiffProtos(
+					actualCreateTimeSeriesReq,
+					expectedCreateTimeSeriesReq,
+				)
+				require.Emptyf(
+					t,
+					diff,
+					"Expected CreateTimeSeries request and actual GCM request differ:\n%v",
+					diff,
+				)
+			})
+
+			t.Run("Real Export", func(t *testing.T) {
```

Comment:
I think you could split this out into a separate file with the integration build rule like you had before if you wanted, rather than use env vars.

Reply:
I also don't have a strong opinion. What do you prefer?

Reply:
Split it into two files, `metrics_test.go` and `metrics_integration_test.go` with build tag